### DISCUSSION TITLE
强编译不过, 是否有必要或考虑支持强编译

### DISCUSSION THREAD

**Discussion by wangzhione** [cloudwu/skynet#1550] [Q&A] [answered]
例如 当前是这样的
`CFLAGS = -g -O2 -Wall -I$(LUA_INC) $(MYCFLAGS)`

当尝试补充成这样的 (最小场景) 

`CFLAGS = -g -O3 -Werror -Wall -Wextra -Wno-missing-field-initializers  -DNDEBUG -I$(LUA_INC) $(MYCFLAGS)`

DNDEBUG 作用在于 Release 模式去掉 assert 宏. 当去掉 assert 时候引发例如 

`
static inline void check_wb_list(struct wb_list * list) {
    assert(list->head == NULL);
    assert(list->tail == NULL);
}
`
引发 -Wextra -Werror 

error: unused parameter ‘list’ [-Werror=unused-parameter]
 static inline void check_wb_list(struct wb_list * list) {

报错. 

你怎么看, 这类严格模式选项编译呢? 是否有计划尝试?

**Comment by cloudwu** [comment, marked as answer]
没这个计划。同时也不建议关闭 assert 。assert 有性能问题了吗？

**Reply by wangzhione** [reply]
没有性能问题, 强迫症习惯 区分 Debug 和 Release

**Reply by cloudwu** [reply]
但是 assert 和 debug 和 release 无关，release 更应该让 assert 工作。因为 release 没有调试信息，出问题时更缺乏定位问题的手段。

**Reply by wangzhione** [reply]
可能这是 window 上使用 Visual Studio 不好习惯. 他在 Release 中默认加了 /D "NDEBUG" 关闭条件编译调试代码开关 (即不编译 assert 函数) . 觉得你说的有道理, 特定时候 **release 更应该让 assert 工作**.

**Reply by sniper00** [reply]
这种设置应该是通用的行为 Cmake应该也是这样
